src/Arcus.Messaging.Abstractions/MessageCorrelationResult.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using Arcus.Messaging.Abstractions.MessageHandling;
+using GuardNet;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+
+namespace Arcus.Messaging.Abstractions
+{
+    /// <summary>
+    /// Represents the correlation result of a received Azure Service Bus message.
+    /// This result will act as the scope of the request telemetry.
+    /// </summary>
```

Comment:
If I'm not mistaken, this class is only useful when using hierarchical logging ?
Should we add this as a `<remark>` ?

Reply:
That's incorrect. Both for W3C as Hierarchical logging this class is used, so as to avoid confusing and have a streamliend approach.